File: app/src/main/java/com/luisenricke/simpleroomapp/database/entity/Pet.java
package com.luisenricke.simpleroomapp.database.entity;

import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import static com.luisenricke.simpleroomapp.database.entity.Pet.COLUMN.*;

@Entity(tableName = TABLE,
        foreignKeys = {@ForeignKey(
                entity = User.class,
                parentColumns = User.COLUMN.ID,
                childColumns = USER,
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE
        )}
)
public class Pet {
    public interface COLUMN {
        String TABLE = "Pet";
        String ID = "id";
        String NAME = "name";
        String BORN_DAY = "born";
        String USER = "user_id";
    }

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = ID)
    private int id;
    @ColumnInfo(name = NAME)
    private String name;
    @ColumnInfo(name = BORN_DAY)
    private String born;
    @ColumnInfo(name = USER)
    private int userId;

    @Ignore
    public Pet() {
    }

    @Ignore
    public Pet(int id, String name, String born, int userId) {
        this.id = id;
        this.name = name;
        this.born = born;
        this.userId = userId;
    }

    @Ignore
    public Pet(String name, String born) {
        this.name = name;
        this.born = born;
    }

    public Pet(String name, String born, int userId) {
        this.name = name;
        this.born = born;
        this.userId = userId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBorn() {
        return born;
    }

    public void setBorn(String born) {
        this.born = born;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || obj.getClass() != Pet.class) return false;
        Pet cast = (Pet) obj;
        return cast.id == getId() && cast.name.equals(getName()) && cast.born.equals(getBorn()) && cast.userId == getUserId();
    }
}

File: app/src/main/java/com/luisenricke/simpleroomapp/data/entity/Pet.kt
package com.luisenricke.simpleroomapp.data.entity

import androidx.room.*
import java.util.*

@Suppress("SpellCheckingInspection")
@Entity(
    tableName = Pet.SCHEMA.TABLE,
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = arrayOf(User.SCHEMA.ID),
            childColumns = arrayOf(Pet.SCHEMA.USER_ID),
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Pet(
    @ColumnInfo(name = SCHEMA.NAME)
    val name: String,
    @ColumnInfo(name = SCHEMA.BORN_DAY)
    val bornDay: Date,
    @ColumnInfo(name = SCHEMA.USER_ID, index = true)
    val userId: Int,
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = SCHEMA.ID)
    val id: Long = 0
) {
    object SCHEMA {
        const val TABLE = "Pet"
        const val ID = "id"
        const val NAME = "name"
        const val BORN_DAY = "born"
        const val USER_ID = "user_id"
    }
}
